src/GraphQL/Execution/IExecutionStrategySelector.cs

```diff
@@ -0,0 +1,13 @@
+using GraphQLParser.AST;
+
+namespace GraphQL.Execution
+{
+    /// <summary>
+    /// Returns an instance of an <see cref="IExecutionStrategy"/> for a specified operation type.
+    /// </summary>
+    public interface IExecutionStrategySelector
+    {
+        /// <inheritdoc cref="IExecutionStrategySelector"/>
+        IExecutionStrategy Select(OperationType operationType);
```

Comment:
I would prefer `ExecutionContext` over `OperationType`. Consider such code:
```
 IExecutionStrategy Select(ExecutionContext context)
{
    if (context.UserContext["forceParallel"] == true)
       return (parallel);
    else
       return (default);
}
```

`ExecutionContext` provides more _context_ for making decision.

Reply:
Fine with me.